src/how-to-build-and-run.md

```diff
@@ -55,13 +55,22 @@ compiler to compile the newer version. In particular, the newer version of the
 compiler, `libstd`, and other tooling may use some unstable features
 internally. The result is the compiling `rustc` is done in stages.
 
-- **Stage 0:** the stage0 compiler is the current _beta_ compiler; we
-  download this binary from the internet.
-- **Stage 1:** the code in your clone is then compiled with the stage
-  0 compiler to produce the stage 1 compiler.
-- **Stage 2:** the code in your clone is then compiled with the stage
-  1 compiler *again* to produce the stage 2 compiler (i.e. it builds
-  itself).
+- **Stage 0:** the stage0 compiler can be your existing
+  (perhaps older version of)
+  Rust compiler, the current _beta_ compiler or you may download the binary
+  from the internet
```

Comment:
Nit: no period at end of sentence

Also, this seems a bit strange... how can you guarantee that the compiler already installed _can_ compile the nightly compiler?

Reply:
AFAIK the incremental bootstrapping process goes like this. Assume you have a language A and it's compiler "CA"
1) You extend A slightly to obtain B
2) You extend the compiler CA (to CB) so that it compiles B, but *without* using the added new features
3) Use CA to compile CB, then you (possibly after rewriting CB with the added features) compile CB with CB ...

[This](https://softwareengineering.stackexchange.com/questions/105313/how-could-the-first-c-compiler-be-written-in-c) helped me understand it *not only for the first compiler* you write for a new language but also how the bootstrap continues after that.